src/ui/components/tags/TagEditor.vue

```diff
@@ -92,57 +88,67 @@ export default {
         }
     },
     watch: {
-        annotation: {
-            handler() {
-                this.tagsChanged(this.annotation.tags);
-            },
-            deep: true
-        },
-        annotationQuery: {
+        annotations: {
             handler() {
-                this.unloadAnnotation();
-                this.loadAnnotation();
+                this.annotationsChanged();
             },
             deep: true
         }
     },
     mounted() {
-        this.loadAnnotation();
+        this.annotationDeletionListener = this.annotationDeletionListener.bind(this);
+        this.annotationsChanged();
     },
     destroyed() {
-        if (this.removeTagsListener) {
-            this.removeTagsListener();
-        }
+        Object.keys(this.deleteAnnotationListeners).forEach(annotationKeyString => {
+            this.deleteAnnotationListeners[annotationKeyString]();
+        });
+        this.deleteAnnotationListeners = [];
     },
     methods: {
-        addAnnotationListener(annotation) {
-            if (annotation && !this.removeTagsListener) {
-                this.removeTagsListener = this.openmct.objects.observe(annotation, '*', (newAnnotation) => {
-                    this.tagsChanged(newAnnotation.tags);
-                    this.annotation = newAnnotation;
-                });
+        annotationsChanged() {
+            this.addAnnotationListeners(this.annotations);
+            if (this.annotations && this.annotations.length) {
+                this.tagsChanged();
             }
         },
-        async loadAnnotation() {
-            this.annotation = await this.openmct.annotation.getAnnotation(this.annotationQuery, this.annotationSearchType);
-            this.addAnnotationListener(this.annotation);
-            if (this.annotation && this.annotation.tags) {
-                this.tagsChanged(this.annotation.tags);
+        annotationDeletionListener(changedAnnotation) {
+            const matchingAnnotation = this.annotations.find((possibleMatchingAnnotation) => {
+                return this.openmct.objects.areIdsEqual(possibleMatchingAnnotation.identifier, changedAnnotation.identifier);
+            });
+            if (matchingAnnotation) {
+                matchingAnnotation._deleted = changedAnnotation._deleted;
```

Comment:
I think we can remove some boilerplate here if we wrap the annotations in a mutable so that this sort of synchronization happens automatically.

The object API [has a function](https://github.com/nasa/openmct/blob/master/src/api/objects/ObjectAPI.js#L515) (currently private but we should make it public) for turning a non-mutable version of an object into a mutable one.

If we were to modify Notebook.vue to wrap all of the annotations in mutables (eg. `this.annotations = this.annotations.map(this.openmct.objects.toMutable)` we could remove some boilerplate here. 

We just need to make sure we destroy the mutables when the notebook view is destroyed.

Reply:
> The object API [has a function](https://github.com/nasa/openmct/blob/master/src/api/objects/ObjectAPI.js#L515) (currently private but we should make it public) for turning a non-mutable version of an object into a mutable one.

I saw that, but was afraid to use it 🙂.

> If we were to modify Notebook.vue to wrap all of the annotations in mutables (eg. this.annotations = this.annotations.map(this.openmct.objects.toMutable) we could remove some boilerplate here.

I'll give this a shot.